### PR TITLE
allow COMP cheat to change complevel in-game

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

File: src/g_game.c [hunk 1]
```diff
@@ -2963,9 +2981,11 @@ void G_ReloadDefaults(void)
   compatibility = false;     // killough 10/98: replaced by comp[] vector
   memcpy(comp, default_comp, sizeof comp);
 
-  demo_version = G_GetWadComplevel();
-
+  if (gamestate != GS_LEVEL)
```
Comment (fabiangreffrath) [inline src/g_game.c:2984]: Not sure about this one. Is there a better/safer way to determine if the function is called from withing a running game (i.e. from the scope of a cheat code)?

Comment (fabiangreffrath) [reply src/g_game.c:2984]: Perhapt check for `leveltime > 0` or `gameaction == ga_nothing` or just add a parameter to the function?

**Comment by rfomin** [comment]
We have some `demo_compatibility` and `demo_version` checks in `p_setup.c`, so can we actually change the complevel in-game? I didn't know about this cheat, it will work in most cases, but I think we should remove it.

**Comment by fabiangreffrath** [comment]
The checks in `p_setup.c` are mostly guards for overflow emulation or slime trail removal. I'd say, if you know what you're doing (and that what you're doing is wrong 😉), it should be possible to load a map in one complevel and finish it in another. 🤷

**Comment by rfomin** [comment]
I can't test it right now, but I think sometimes this cheat can break the game. Okay, but we should at least put a disclaimer in the changelog.

**Comment by fabiangreffrath** [comment]
ALright!

File: src/g_game.c (hunk 1 above)
Comment (rfomin) [reply src/g_game.c:2984]: I think function parameter is more readable/simple.

Comment (kraflab) [reply src/g_game.c:2984]: This may interest you: https://github.com/kraflab/dsda-doom/commit/0092c134af94c6d26b4c4fa99e4d67600f5ddaaa

Comment (kraflab) [reply src/g_game.c:2984]: Ah, you're looking in G_ReloadDefaults, not so easy. 😄 
Incidentally one of my next refactoring tasks is to fix what I think is chaos in how the source manages all these game configuration details, but that's for another day :^)

**Comment by fabiangreffrath** [comment]
> ... but I think we should remove it.

There I quite some other cheat code that I consider as pointless, e.g. "ice", "push", "nuke", etc. We should really consider removing these.

**Comment by rfomin** [comment]
> There I quite some other cheat code that I consider as pointless, e.g. "ice", "push", "nuke", etc. We should really consider removing these.

Sure. Even [doomiki](https://doomwiki.org/wiki/Source_port_cheat_codes) don't know about them, I doubt anyone has ever used them.

**Comment by fabiangreffrath** [comment]
Seems to work. :shrug:

**Review by rfomin — APPROVED** [review]
LGTM, though I haven't tested it yet.

**Comment by fabiangreffrath** [comment]
> LGTM, though I haven't tested it yet.

Changing compatibility level in-game, what could probably go wrong? :stuck_out_tongue_winking_eye:

**Comment by fabiangreffrath** [comment]
> LGTM, though I haven't tested it yet.

Seriously, any idea for a real-life test for this cheat? Maybe the Barons at the end of Whispers of Satan MAP06 as Lila Feuer mentioned in the DW forums? Like, start the map with complevel Vanilla and switch to Boom right before the exit?

**Comment by fabiangreffrath** [comment]
Just checked in the last section of WOS.wad MAP06. Hitting the switches to the left and right of the exit portal should lower the bars right in front of it, which only peroperly works in Boom complevel. I have a nomonsters savegame of my player standing right between the switches and if I switch them with complevel MBF21 the bars won't lower. But if I enter COMP09 right before, they will lower and the exit portal becomes accessible. :partying_face:

**Pull request merged**